src/main/java/org/tron/common/overlay/portmap/router/cling/ClingRegistryListener.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * UPnP PortMapper - A tool for managing port forwardings via UPnP Copyright (C) 2015 Christoph
+ * Pirkl <christoph at users.sourceforge.net>
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.  If
+ * not, see <http://www.gnu.org/licenses/>.
+ */
+package org.tron.common.overlay.portmap.router.cling;
+
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
+import org.fourthline.cling.model.meta.Device;
+import org.fourthline.cling.model.meta.Service;
+import org.fourthline.cling.model.types.DeviceType;
+import org.fourthline.cling.model.types.ServiceType;
+import org.fourthline.cling.model.types.UDADeviceType;
+import org.fourthline.cling.model.types.UDAServiceType;
+import org.fourthline.cling.registry.DefaultRegistryListener;
+import org.fourthline.cling.registry.Registry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClingRegistryListener extends DefaultRegistryListener {
+
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  public static final DeviceType IGD_DEVICE_TYPE = new UDADeviceType("InternetGatewayDevice", 1);
+  public static final DeviceType CONNECTION_DEVICE_TYPE = new UDADeviceType("WANConnectionDevice",
+      1);
+
+  public static final ServiceType IP_SERVICE_TYPE = new UDAServiceType("WANIPConnection", 1);
+  public static final ServiceType PPP_SERVICE_TYPE = new UDAServiceType("WANPPPConnection", 1);
+
+  private final SynchronousQueue<Service<?, ?>> foundServices;
+
+  public ClingRegistryListener() {
+    this.foundServices = new SynchronousQueue<>();
+  }
+
+  public Service<?, ?> waitForServiceFound(final long timeout, final TimeUnit unit) {
+    try {
+      return foundServices.poll(timeout, unit);
+    } catch (final InterruptedException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1583&issues=AWZcuxrN9tx2us2yFA5e&open=AWZcuxrN9tx2us2yFA5e)

[ISSUE_KEY]: AWZcuxrN9tx2us2yFA5e